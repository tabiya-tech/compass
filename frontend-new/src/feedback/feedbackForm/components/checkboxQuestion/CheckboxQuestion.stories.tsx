import { Meta, StoryObj } from "@storybook/react";
import CheckboxQuestion from "src/feedback/feedbackForm/components/checkboxQuestion/CheckboxQuestion";
import { action } from "@storybook/addon-actions";

const meta: Meta<typeof CheckboxQuestion> = {
  title: "Feedback/CheckboxQuestion",
  component: CheckboxQuestion,
  tags: ["autodocs"],
  args: {
    notifyChange: (selectedOptions, comments) => {
      action("notifyChange")(selectedOptions, comments);
    },
  },
};

export default meta;

type Story = StoryObj<typeof CheckboxQuestion>;

export const Shown: Story = {
  args: {
    questionId: "accuracy_relevance",
    questionText: "How accurate and relevant was the information provided?",
    options: [
      { key: "accurate", value: "Accurate" },
      { key: "relevant", value: "Relevant" },
      { key: "upToDate", value: "Up-to-date" },
      { key: "easyToUnderstand", value: "Easy to understand" },
    ],
    selectedOptions: [],
  },
};
