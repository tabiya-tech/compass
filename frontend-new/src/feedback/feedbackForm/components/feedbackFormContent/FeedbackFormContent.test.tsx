// mute the console
import "src/_test_utilities/consoleMock";

import { render, screen } from "src/_test_utilities/test-utils";
import { fireEvent } from "@testing-library/react";
import { DATA_TEST_ID as CUSTOM_RATING_DATA_TEST_ID } from "src/feedback/feedbackForm/components/customRating/CustomRating";
import { DATA_TEST_ID as YES_NO_DATA_TEST_ID } from "src/feedback/feedbackForm/components/yesNoQuestion/YesNoQuestion";
import FeedbackFormContent, {
  DATA_TEST_ID,
} from "src/feedback/feedbackForm/components/feedbackFormContent/FeedbackFormContent";
import { DATA_TEST_ID as CHECKBOX_DATA_TEST_ID } from "src/feedback/feedbackForm/components/checkboxQuestion/CheckboxQuestion";
import stepsContent from "src/feedback/feedbackForm/stepsContent";
import { DATA_TEST_ID as COMMENT_TEXT_FIELD_TEST_ID } from "src/feedback/feedbackForm/components/commentTextField/CommentTextField";

describe("FeedbackFormContent", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
  });

  test("should render component successfully", () => {
    // GIVEN the component
    const givenFeedbackFormContent = <FeedbackFormContent notifySubmit={jest.fn()} />;

    // WHEN the component is rendered
    render(givenFeedbackFormContent);

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the feedback form content to be in the document
    const feedbackFormContent = screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_CONTENT);
    expect(feedbackFormContent).toBeInTheDocument();
    // AND the feedback form content title to be in the document
    expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_CONTENT_TITLE)).toBeInTheDocument();
    // AND the feedback form content divider to be in the document
    expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_CONTENT_DIVIDER)).toBeInTheDocument();
    // AND the feedback form content questions to be in the document
    expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_CONTENT_QUESTIONS)).toBeInTheDocument();
    // AND the feedback form next button to be in the document
    expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_NEXT_BUTTON)).toBeInTheDocument();
    // AND the feedback form back button to be in the document
    expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_BACK_BUTTON)).toBeInTheDocument();
    // AND to match the snapshot
    expect(feedbackFormContent).toMatchSnapshot();
  });

  describe("action tests", () => {
    test("should call handleNext when next button is clicked", () => {
      // GIVEN the component
      const givenFeedbackFormContent = <FeedbackFormContent notifySubmit={jest.fn()} />;
      // AND the component is rendered
      render(givenFeedbackFormContent);

      // WHEN the next button is clicked
      const nextButton = screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_NEXT_BUTTON);
      fireEvent.click(nextButton);

      // THEN expect to go to the next step
      expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_CONTENT_TITLE)).toHaveTextContent(stepsContent[1].label);
    });

    test("should call handlePrevious when back button is clicked", () => {
      // GIVEN the component
      const givenFeedbackFormContent = <FeedbackFormContent notifySubmit={jest.fn()} />;
      // AND the component is rendered
      render(givenFeedbackFormContent);

      // WHEN the next button is clicked to move to the next step
      const nextButton = screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_NEXT_BUTTON);
      fireEvent.click(nextButton);
      // AND the back button is clicked
      const backButton = screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_BACK_BUTTON);
      fireEvent.click(backButton);

      // THEN expect to go to the previous step
      expect(screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_CONTENT_TITLE)).toHaveTextContent(stepsContent[0].label);
    });

    test("should call handleSubmit with exact answer data when answering each step", () => {
      // GIVEN the FeedbackFormContent component
      const mockHandleSubmit = jest.fn();
      const givenFeedbackFormContent = <FeedbackFormContent notifySubmit={mockHandleSubmit} />;
      // AND the component is rendered
      render(givenFeedbackFormContent);

      // WHEN on the first step, answer rating question
      const starRating = screen.getAllByTestId(CUSTOM_RATING_DATA_TEST_ID.CUSTOM_RATING_ICON)[4];
      fireEvent.click(starRating);

      // AND move to next step
      const nextButton = screen.getByTestId(DATA_TEST_ID.FEEDBACK_FORM_NEXT_BUTTON);
      fireEvent.click(nextButton);

      // AND on the second step, select a checkbox option
      const checkboxInput = screen.getAllByTestId(CHECKBOX_DATA_TEST_ID.CHECKBOX_OPTION)[0];
      fireEvent.click(checkboxInput);

      // AND move to next step
      fireEvent.click(nextButton);

      // AND on the third step, answer the yes/no question
      const yesNoInput = screen.getAllByTestId(YES_NO_DATA_TEST_ID.RADIO_YES)[1];
      fireEvent.click(yesNoInput);

      // AND move to next step
      fireEvent.click(nextButton);

      // AND on the last step, provide a custom rating comment
      const customRatingInput = screen.getAllByTestId(CUSTOM_RATING_DATA_TEST_ID.CUSTOM_RATING_ICON)[8];
      fireEvent.click(customRatingInput);


      // AND submit the form
      fireEvent.click(nextButton);

      // THEN expect mockHandleSubmit to have been called exactly once
      expect(mockHandleSubmit).toHaveBeenCalledTimes(1);

      // AND expect the exact answer data structure
      const submittedAnswers = mockHandleSubmit.mock.calls[0][0];
      expect(submittedAnswers).toHaveLength(4);

      // First step answer (custom rating)
      expect(submittedAnswers[0]).toEqual({
        question_id: stepsContent[0].questions[0].questionId,
        answer: {
          comment: "",
          rating_boolean: undefined,
          rating_numeric: 5,
          selected_options: undefined,
        },
        is_answered: true,
      });

      // Second step answer (checkbox)
      expect(submittedAnswers[1]).toEqual({
        question_id: stepsContent[1].questions[1].questionId,
        answer: {
          comment: "",
          rating_boolean: undefined,
          rating_numeric: undefined,
          selected_options: [stepsContent[1].questions[1].options![0].key],
        },
        is_answered: true,
      });

      // Third step answer (yes/no)
      expect(submittedAnswers[2]).toEqual({
        question_id: stepsContent[2].questions[1].questionId,
        answer: {
          comment: "",
          rating_boolean: true,
          rating_numeric: undefined,
          selected_options: undefined,
        },
        is_answered: true,
      });

      // Last step answer (custom rating)
      expect(submittedAnswers[3]).toEqual({
        question_id: stepsContent[3].questions[0].questionId,
        answer: {
          comment: "",
          rating_boolean: undefined,
          rating_numeric: 9,
          selected_options: undefined,
        },
        is_answered: true,
      });
    });

    test("should call handleAnswerChange when a question is answered", () => {
      // GIVEN the FeedbackFormContent component
      const givenFeedbackFormContent = <FeedbackFormContent notifySubmit={jest.fn()} />;
      // AND the component is rendered
      render(givenFeedbackFormContent);

      // WHEN question is answered
      const customRating = screen.getAllByTestId(CUSTOM_RATING_DATA_TEST_ID.CUSTOM_RATING_ICON)[4];
      fireEvent.click(customRating);

      const input = screen.getByTestId(COMMENT_TEXT_FIELD_TEST_ID.COMMENT_TEXT_FIELD);
      fireEvent.change(input, { target: { value: "This is a comment" } });

      // THEN expect the answer to be saved
      expect(input).toHaveValue("This is a comment");
    });
  });
});
