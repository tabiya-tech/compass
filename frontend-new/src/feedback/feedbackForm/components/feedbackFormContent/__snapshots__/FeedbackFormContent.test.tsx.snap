// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackFormContent should render component successfully 1`] = `
<div
  class="MuiBox-root css-11artbl"
  data-testid="feedback-form-content-07d1ed9d-6ced-4a24-912a-9b89199df67f"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1lysc6y-MuiTypography-root"
    data-testid="feedback-form-content-title-07d1ed9d-6ced-4a24-912a-9b89199df67f"
  >
    Overall Satisfaction
  </p>
  <div
    class="MuiBox-root css-15bovrc"
    data-testid="feedback-form-content-questions-07d1ed9d-6ced-4a24-912a-9b89199df67f"
  >
    <div
      class="MuiBox-root css-ifx08d"
      data-testid="steps-component-3bf900a2-f9fa-4b28-8c20-8bc21570635c"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-yj95vr"
          data-testid="custom-rating-container-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
        >
          <div
            class="MuiBox-root css-1w00lo4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1au4tcm-MuiTypography-root"
              data-testid="question-text-bf2b8e40-5a77-4eac-a0ea-ba8af97a5d67"
            >
              How easy or difficult was it to interact with Compass and understand its responses?
            </p>
            <div
              class="MuiBox-root css-a0ovpn"
            >
              <span
                class="MuiRating-root MuiRating-sizeSmall css-cod5a2-MuiRating-root"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r1:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="1"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    1 Star
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r1:"
                  name="interaction_ease"
                  type="radio"
                  value="1"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r2:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="2"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    2 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r2:"
                  name="interaction_ease"
                  type="radio"
                  value="2"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r3:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="3"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    3 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r3:"
                  name="interaction_ease"
                  type="radio"
                  value="3"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r4:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="4"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    4 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r4:"
                  name="interaction_ease"
                  type="radio"
                  value="4"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r5:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="5"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r5:"
                  name="interaction_ease"
                  type="radio"
                  value="5"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r6:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="6"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    6 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r6:"
                  name="interaction_ease"
                  type="radio"
                  value="6"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":r7:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="7"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    7 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":r7:"
                  name="interaction_ease"
                  type="radio"
                  value="7"
                />
                <label
                  class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
                >
                  <input
                    checked=""
                    class="MuiRating-visuallyHidden"
                    id="interaction_ease-empty"
                    name="interaction_ease"
                    type="radio"
                    value=""
                  />
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    Empty
                  </span>
                </label>
              </span>
              <div
                class="MuiBox-root css-jyedia"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-bwgvvh-MuiTypography-root"
                  data-testid="custom-rating-low-label-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                >
                  Difficult
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-bwgvvh-MuiTypography-root"
                  data-testid="custom-rating-high-label-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                >
                  Easy
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-vg49q3-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-voxow3-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-cmpe59-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="comment-text-field-ab74a074-67f6-4a8a-b2b2-a582d1d56c9c"
                id=":r8:"
                placeholder="Please explain your rating and share any suggestions for improving the ease of interaction with Compass."
                rows="3"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-cmpe59-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-yj95vr"
          data-testid="custom-rating-container-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
        >
          <div
            class="MuiBox-root css-1w00lo4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1au4tcm-MuiTypography-root"
              data-testid="question-text-bf2b8e40-5a77-4eac-a0ea-ba8af97a5d67"
            >
              How satisfied are you with Compass?
            </p>
            <div
              class="MuiBox-root css-a0ovpn"
            >
              <span
                class="MuiRating-root MuiRating-sizeSmall css-cod5a2-MuiRating-root"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":ra:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="1"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    1 Star
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":ra:"
                  name="satisfaction_with_compass"
                  type="radio"
                  value="1"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rb:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="2"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    2 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rb:"
                  name="satisfaction_with_compass"
                  type="radio"
                  value="2"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rc:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="3"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    3 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rc:"
                  name="satisfaction_with_compass"
                  type="radio"
                  value="3"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rd:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="4"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    4 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rd:"
                  name="satisfaction_with_compass"
                  type="radio"
                  value="4"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":re:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    data-testid="custom-rating-icon-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                    value="5"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1mccn0v-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":re:"
                  name="satisfaction_with_compass"
                  type="radio"
                  value="5"
                />
                <label
                  class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
                >
                  <input
                    checked=""
                    class="MuiRating-visuallyHidden"
                    id="satisfaction_with_compass-empty"
                    name="satisfaction_with_compass"
                    type="radio"
                    value=""
                  />
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    Empty
                  </span>
                </label>
              </span>
              <div
                class="MuiBox-root css-jyedia"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-bwgvvh-MuiTypography-root"
                  data-testid="custom-rating-low-label-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                >
                  Unsatisfied
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-bwgvvh-MuiTypography-root"
                  data-testid="custom-rating-high-label-7bb1da6e-bd0f-4edf-bbbb-ff7ade168944"
                >
                  Satisfied
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-4rn412-MuiDivider-root"
    color="primary"
    data-testid="feedback-form-content-divider-07d1ed9d-6ced-4a24-912a-9b89199df67f"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic css-ge39m6-MuiPaper-root-MuiMobileStepper-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-1kyjyn3-MuiButtonBase-root-MuiButton-root"
      data-testid="feedback-form-back-button-07d1ed9d-6ced-4a24-912a-9b89199df67f"
      disabled=""
      tabindex="-1"
      type="button"
    >
      Previous
    </button>
    <div
      class="MuiMobileStepper-dots css-1384nzf-MuiMobileStepper-dots"
    >
      <div
        class="MuiMobileStepper-dot MuiMobileStepper-dotActive css-15nuweu-MuiMobileStepper-dot"
      />
      <div
        class="MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot"
      />
      <div
        class="MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot"
      />
      <div
        class="MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot"
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-1j5hbp5-MuiButtonBase-root-MuiButton-root"
      data-testid="feedback-form-next-button-07d1ed9d-6ced-4a24-912a-9b89199df67f"
      style="width: 100px;"
      tabindex="0"
      type="button"
    >
      Next
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
